src/main/java/com/matera/cadastrocentral/telephone/TelephoneService.java

```diff
@@ -0,0 +1,118 @@
+package com.matera.cadastrocentral.telephone;
+
+import javassist.tools.web.BadHttpRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+@Service
+public class TelephoneService {
+
+    @Autowired
+    public TelephoneService(TelephoneRepository telephoneRepository) {
+        this.telephoneRepository = telephoneRepository;
+    }
+
+    private final TelephoneRepository telephoneRepository;
+
+    public List<Telephone> findAllTelephonesByClientId(UUID clientId) {
+        List<Telephone> phoneList = telephoneRepository.findAllByClientId(clientId);
+        if (phoneList.isEmpty()) {
+            throw new ClientNotFound();
+        } else {
+            return phoneList;
+        }
+    }
+
+    public Optional<Telephone> findByTelephoneId(UUID telephoneId) {
+        Optional<Telephone> optionalTelephone = telephoneRepository.findById(telephoneId);
+        if (optionalTelephone.isPresent()) {
+            return optionalTelephone;
+        } else {
+            throw new TelephoneNotFound();
+        }
+    }
+
+    public Telephone insertTelephone(UUID clientId, TelephoneDTO telephone) throws TelephoneAlreadyExists {
+        // Checks if the input already exists in database before inserting
+        List<Telephone> phoneList = telephoneRepository.findAllByClientId(clientId);
```

Comment:
GGreat initiative here, on checking if the telefone number already exists :)

it can be simplified by implementing the "equals" method on Telephone class, create  a new Telephone object from the DTO, and checking if telephone exist by doing a "phoneList.contains(<Telephone object created from DTO class>)"

Reply:
Thanks, I'll try to make it simpler :smile: